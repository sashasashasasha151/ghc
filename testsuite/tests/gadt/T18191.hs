{-# LANGUAGE GADTs #-}
{-# LANGUAGE RankNTypes #-}
module T18191 where

data T where
  MkT :: (forall a. a -> b -> T)

data S a where
  MkS :: (forall a. S a)

data U a where
  MkU :: (Show a => U a)
