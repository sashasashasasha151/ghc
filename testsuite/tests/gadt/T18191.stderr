
T18191.hs:6:11: error:
    GADT constructor type signature cannot contain nested ‘forall’s or contexts
    In the definition of data constructor ‘MkT’

T18191.hs:9:11: error:
    GADT constructor type signature cannot contain nested ‘forall’s or contexts
    In the definition of data constructor ‘MkS’

T18191.hs:12:11: error:
    GADT constructor type signature cannot contain nested ‘forall’s or contexts
    In the definition of data constructor ‘MkU’
